### PR TITLE
phpstan level 2

### LINKED ISSUE #14647: phpstan - level 2
https://github.com/phpmyadmin/phpmyadmin/blob/d4b4124a5e5d4d3014e41981ae1b0c548c619fd1/libraries/classes/CentralColumns.php#L1043-L1079

At line `1058` it returns an array, this will fail because typehint is set to `int`

### Report
```
 ------ ------------------------------------------------------------------------- 
  Line   /mnt/Dev/phpmyadmintest/libraries/classes/CentralColumns.php             
 ------ ------------------------------------------------------------------------- 
  1054   PHPDoc tag @return with type array is incompatible with native type int  
 ------ ------------------------------------------------------------------------- 
```

What needs to be returned ?

### PR DESCRIPTION
**Pull request by williamdes**
### Description

Fix errors found by phpstan.

Upgraded to `level=2`
Removed failure allowed from CI.

Fixes #14647

### CODE REVIEW

**Review by MauricioFauth — APPROVED** [review]

File: libraries/classes/Dbi/DbiMysqli.php [hunk 1]
```diff
@@ -532,30 +534,35 @@ public function fieldName($result, $i)
...
+            // Unfortunately there is no equivalent in the mysql extension.
+            if (($type == MYSQLI_TYPE_TINY_BLOB || $type == MYSQLI_TYPE_BLOB
+                || $type == MYSQLI_TYPE_MEDIUM_BLOB || $type == MYSQLI_TYPE_LONG_BLOB
+                || $type == MYSQLI_TYPE_VAR_STRING || $type == MYSQLI_TYPE_STRING)
+                && 63 == $charsetNumber
+            ) {
+                $flags[] = 'binary';
+            }
+            return implode(' ', $flags);
+        } else {
```
[the remark this answers is not in the document]
Comment (williamdes) [reply libraries/classes/Dbi/DbiMysqli.php]: From php docs: http://php.net/manual/en/mysqli-result.fetch-field-direct.php
> Returns an object which contains field definition information or FALSE if no field information for specified fieldnr is available.

I am not quite sure if the else should be removed ?

**Comment by williamdes** [comment]
@mauriciofauth I rebased onto 3314afbd85c687a7191b8044f0686e49b28cef0e
Do you want me to group my commits before merge ?
You can merge if everything looks good for you !
Last commit added after rebase : 37db663f73073071fd7e2bd5da4dbb1742e73ff4

**Comment by mauriciofauth** [comment]
@williamdes Can you squash your commits before you merge this PR? Some commit messages doesn't mean much.

**Comment by williamdes** [comment]
@mauriciofauth Done !
As soon as tests pass I will merge :)

**Pull request merged**